cms/static/cms/js/modules/cms.base.js

```diff
@@ -39,6 +39,8 @@ var CMS = {
     $(document).ready(function () {
         CMS.API.Helpers = {
 
+            isTouch: window.ontouchstart !== undefined,
```

Comment:
modernizr officially says that it it not possible to detect touch device, just the availablity of touch events

it comes from pointerover of course, but we can at least detect if there was a touch recently / nearby, and trigger some flag to change toolbar to the touch mode, and then after showing it - reset the flag. and keep doing it over and over again

(that's roughly what i was thinking)